### PR TITLE
[Fix #74] Make sure strings are Unicode in the restapi

### LINKED ISSUE #74: Make sure strings are Unicode in the restapi
When they are read inside restapi.py, incoming strings are in the 'bytes' format.
What I call incoming strings are e.g. 

    u = objects.Profile()
    u.name = request.args["name"][0]

Protobuf's type checker will check whether ```request.args["name"][0]``` if of ```bytes``` type. When it detects that it is, it fails when the string contains non-ascii characters, like 'é' with the following exception : "has type bytes, but isn\'t in 7-bit ASCII 'encoding. Non-ASCII strings must be converted to unicode objects before being added.".
That prevents entering any non-ascii description everywhere.

### PR DESCRIPTION
**Pull request by jojva**

### CODE REVIEW

File: api/restapi.py [hunk 1]
```diff
@@ -368,8 +368,8 @@ def add_social_account(self, request):
         try:
             p = Profile(self.db)
             if "account_type" in request.args and "username" in request.args and "proof" in request.args:
-                p.add_social_account(request.args["account_type"][0], request.args["username"][0],
-                                     request.args["proof"][0] if "proof" in request.args else None)
+                p.add_social_account(request.args["account_type"][0], request.args["username"][0].decode('utf8'),
+                                     request.args["proof"][0].decode('utf8') if "proof" in request.args else None)
```
Comment (gasull) [inline api/restapi.py]: The guideline say nothing gets merged if it isn't PEP-8 compliant, and these lines are over 80 characters.

Nitpick: you're mixing single and double quotes.

Maybe this instead:
```Python
                if "proof" in request.args:
                    proof = request.args["proof"][0].decode("utf8")
                else:
                    proof = None
                p.add_social_account(
                    request.args["account_type"][0], 
                    request.args["username"][0].decode('utf8'),
                    proof,
                )
```

Comment (hoffmabc) [inline api/restapi.py]: Great point and appreciate that contribution.

Comment (jojva) [inline api/restapi.py]: Updated. Please note that there are:
* 19 E402: module level import not at top of file
* 105 E501 line too long

in that single file

**Comment by gasull** [comment]
This LGTM, but I am still new to the code.  Maybe someone else could take another look.

**Comment by jojva** [comment]
Hm, and for some reason Travis fails now, I'll have to look into that.

**Comment by cpacia** [comment]
I will check this out soon. Is this the only place with issues or are there others?

**Comment by jojva** [comment]
I only looked at restapi.py

**Comment by cpacia** [comment]
I added this

**Pull request closed without merging**